sdc/tests/test_dataframe.py

```diff
@@ -1473,30 +1482,67 @@ def test_impl(df, df2):
         df = pd.DataFrame({'A': np.arange(n), 'B': np.arange(n)**2})
         df2 = pd.DataFrame({'A': np.arange(n), 'C': np.arange(n)**2})
         df2.A[n // 2:] = n
+        hpat_func(df, df2)
         pd.testing.assert_frame_equal(hpat_func(df, df2), test_impl(df, df2))
 
-    @unittest.skip("needs dict typing in Numba")
+    @skip_sdc_jit
+    def test_isin_df2(self):
+        def test_impl(df, df2):
+            return df.isin(df2)
+
+        cfunc = self.jit(test_impl)
+        df = pd.DataFrame({'num_legs': [2, 4], 'num_wings': [2, 0]}, index=['falcon', 'dog'])
+        df2 = pd.DataFrame({'num_legs': [8, 2], 'num_wings': [0, 2]}, index=['spider', 'falcon'])
+        pd.testing.assert_frame_equal(cfunc(df, df2), test_impl(df, df2))
+
+    @skip_sdc_jit
     def test_isin_dict1(self):
         def test_impl(df):
-            vals = {'A': [2, 3, 4], 'C': [4, 5, 6]}
+            vals = {'A': (2., 3., 4.), 'C': (4., 5., 6.)}
             return df.isin(vals)
 
         hpat_func = self.jit(test_impl)
         n = 11
         df = pd.DataFrame({'A': np.arange(n), 'B': np.arange(n)**2})
         pd.testing.assert_frame_equal(hpat_func(df), test_impl(df))
 
-    @skip_numba_jit
-    def test_isin_list1(self):
+    @skip_sdc_jit
+    def test_isin_ser1(self):
         def test_impl(df):
-            vals = [2, 3, 4]
+            vals = pd.Series([2, 3, 4])
             return df.isin(vals)
 
         hpat_func = self.jit(test_impl)
         n = 11
         df = pd.DataFrame({'A': np.arange(n), 'B': np.arange(n)**2})
         pd.testing.assert_frame_equal(hpat_func(df), test_impl(df))
 
+    @skip_sdc_jit
+    def test_isin_ser2(self):
+        def test_impl(df):
+            vals = pd.Series([2, 3, 4], index=[3, 4, 5])
+            return df.isin(vals)
+
+        hpat_func = self.jit(test_impl)
+        n = 11
+        df = pd.DataFrame({'A': np.arange(n), 'B': np.arange(n) ** 2}, index=np.arange(n) + 1)
+        pd.testing.assert_frame_equal(hpat_func(df), test_impl(df))
+
+    def test_isin(self):
+        def test_impl(df, vals):
+            return df.isin(vals)
+
+        hpat_func = self.jit(test_impl)
+        n = 11
+        df = pd.DataFrame({'A': np.arange(n), 'B': np.arange(n) ** 2})
+        df2 = pd.DataFrame({'A': np.arange(n), 'C': np.arange(n) ** 2})
+        df2.A[n // 2:] = n
+        values = [{1, 2, 5, 7, 8}, [2, 3, 4], pd.DataFrame({}), df2]
```

Comment:
Combining values of types that have separate implementations is not good. I would suggest separate this test into two: first for list and set which fall into iter_impl and second that tests isin_df.